### ISSUE TITLE
It is possible to assign an non-existent enum values

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#33284]
**Godot version:**
3.2 alpha3

**Issue description:**
If you create an enum, you can index it with values that don't exist. I'd expect that there's some parse/compile error when this happens.

**Steps to reproduce:**
1. Write this code
```
enum MY_ENUM{VALUE1, VALUE2}

func _ready() -> void:
	var a = MY_ENUM.VALUE3000
```
2. There's no error until you run the game

**Comment by Calinou** [comment]
Out of curiosity, was an error thrown in previous versions?

**Comment by KoBeWi** [comment]
Nope, it was always like that. It has probably something to do with the fact that enums are just dictionaries. The only difference is that they support autocompletion.

**Comment by bojidar-bg** [comment]
Given that #25103 is still in existence, the code above is will not necessary result in a run time error.
Consider another script which has the following code:
```
func _ready():
	var x = preload("myscript.gd")
	x.MY_ENUM.VALUE3000 = 3000
	add_child(x.new())
```

It will compile and run without errors currently, but in an ideal world it should both error at run time and at compile time.

**Comment by KoBeWi** [comment]
Fixed in 4.0
